### PR TITLE
[Do not merge] RHIROS-502 api_version endpoint

### PR DESCRIPTION
**Pull request by srdeotarse**
## api_version endpoint :boom:

[RHIROS-502](https://issues.redhat.com/browse/RHIROS-502) [BE] | API Version Endpoint

Add api_version endpoint to return reponse -

branch:
commit_sha:
commit_timestamp:
api_version:
deployed_at: 

### Endpoint: 

``/api/ros/v1/api_version``

Feel free to remove this section from PR description once done.

## Why do we need this change? :thought_balloon:

Please include the __context of this change__ here.

## Documentation update? :memo:

- [ ] Yes
- [x] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added
- [ ] DB Migration Added

## Additional :mega:

Feel free to add any other relevant details such as __links, notes, screenshots__, here.

### CODE REVIEW

File: clowdapp.yaml [hunk 1]
```diff
@@ -74,6 +74,16 @@ objects:
...
+          - name: GITHUB_OWNER
+            value: "RedHatInsights"
+          - name: GITHUB_REPO
+            value: "ros-backend"
+          - name: GITHUB_DEPLOYMENT_BRANCH
+            value: "main"
+          - name: GITHUB_API_PAGE_SIZE
+            value: "5"
+          - name: GITHUB_API_MAXIMUM_RETRIES
+            value: "3"
```
Comment (saltgen) [inline clowdapp.yaml:86]: You'll need to define these as parameters at the end of this file, like so

```
- description: organisation/owner-name of the GitHub repo
  name: GITHUB_OWNER
  value: "RedHatInsights"
```

Then use those here,

```
- name: GITHUB_OWNER
  value: ${GITHUB_OWNER}
```
I suggest adding these after `Feature flags` vars in a new section at the end.

File: ros/api/v1/api_version.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from flask_restful import Resource
+from flask import jsonify
```
Comment (saltgen) [inline ros/api/v1/api_version.py:2]: As per design guidelines, standard package import should be on top, 

```
from flask import jsonify
```

Then third party imports 

```
from flask_restful import Resource
from k8s_deployment_status import DeploymentStatus
```

And finally app imports

```
from ros.extensions import metrics
from ros.lib.config import API_VERSION
```
**Note:** These are not formal PEP8 rules, but are common best practices and conventions.

File: ros/api/routes.py [hunk 3]
```diff
@@ -10,17 +10,20 @@
...
+    api.add_resource(IsROSConfiguredApi, f'/api/ros/{API_VERSION}/is_configured')
+    api.add_resource(HostsApi, f'/api/ros/{API_VERSION}/systems')
+    api.add_resource(HostHistoryApi, f'/api/ros/{API_VERSION}/systems/<host_id>/history')
+    api.add_resource(RecommendationsApi, f'/api/ros/{API_VERSION}/systems/<host_id>/suggestions')
+    api.add_resource(HostDetailsApi, f'/api/ros/{API_VERSION}/systems/<host_id>')
+    api.add_resource(RecommendationRatingsApi, f'/api/ros/{API_VERSION}/rating')
+    api.add_resource(OpenAPISpec, f'/api/ros/{API_VERSION}/openapi.json')
+    api.add_resource(CallToActionApi, f'/api/ros/{API_VERSION}/call_to_action')
+    api.add_resource(ExecutiveReportAPI, f'/api/ros/{API_VERSION}/executive_report')
+    api.add_resource(ApiVersion, f'/api/ros/{API_VERSION}/api_version')
```
Comment (saltgen) [inline ros/api/routes.py:29]: @kgaikwad Would like to know your thoughts on global value for `API_VERSION`, would this become a challenge if 
endpoints get versioned differently in future?

**Review by saltgen — CHANGES_REQUESTED** [review]

File: clowdapp.yaml [hunk 4]
```diff
@@ -360,3 +370,20 @@ parameters:
 - description: disable Unleash (feature flags), defaulting to fallback values
   name: BYPASS_UNLEASH
   value: 'false'
+
+# api_deployment_version endpoint
```
Comment (saltgen) [inline clowdapp.yaml:374]:
```suggestion
# Deployment version
```

**Review by saltgen — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 5]
```diff
@@ -28,6 +28,7 @@ boto3 = ">=1.16.13"
 botocore = ">=1.19.13"
 flask-caching = "*"
 flask-unleash = "^2.0.0"
+k8s-deployment-status = "^0.0.6"
```
Comment (r14chandra) [inline pyproject.toml:31]: You can add this package in the [Github action](https://github.com/RedHatInsights/ros-backend/blob/main/.github/workflows/build_with_latest_direct_deps.yml) file for direct dependency as tests are failing for it.

**Review by r14chandra — CHANGES_REQUESTED** [review]

File: ros/api/v1/api_deployment_version.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+from k8s_deployment_status import DeploymentStatus
+from ros.extensions import metrics
+from ros.lib.config import API_VERSION
+
+
+class ApiDeploymentVersion(Resource):
+    @metrics.do_not_track()
+    def get(self):
+        import pdb
+        pdb.set_trace()
```
Comment (upadhyeammit) [inline ros/api/v1/api_deployment_version.py:12]: Remove pdb :D

File: ros/lib/config.py [hunk 7]
```diff
@@ -157,3 +157,4 @@ def build_endpoint_url(ep):
 CW_LOGGING_FORMAT = '%(asctime)s - %(levelname)s  - %(funcName)s - %(message)s'
 ROS_PROCESSOR_PORT = int(os.getenv("ROS_PROCESSOR_PORT", "8000"))
 ROS_API_PORT = int(os.getenv("ROS_API_PORT", "8000"))
+API_VERSION = 'v1'
```
Comment (upadhyeammit) [inline ros/lib/config.py:160]: In future we may get more than one api version and both could be supported? can we depend on api paths maybe to get verion and actually show all deployed api versions?

File: .github/workflows/build_with_latest_direct_deps.yml [hunk 8]
```diff
@@ -99,6 +99,7 @@ jobs:
         botocore = "*"
         flask-caching = "*"
         flask-unleash = "*"
+        k8s-deployment-status = "^0.0.6"
```
Comment (r14chandra) [inline .github/workflows/build_with_latest_direct_deps.yml:102]:
```suggestion
        k8s-deployment-status = "*"
```

**Review by r14chandra — CHANGES_REQUESTED** [review]

**Comment by r14chandra** [comment]
\retest

**Review by kgaikwad — CHANGES_REQUESTED** [review]
@srdeotarse  - PR facing few conflicts. Please have a look.

**Comment by kgaikwad** [comment]
Closing this PR as I don't not see any pr updates since long time.

**Pull request closed without merging**